applications/models.py

```diff
@@ -220,6 +221,38 @@ def get_notification_context(self):
             "application": self,
         }
 
+    def _validate_status(self):
+        """
+        If the BerthApplication does not have a BerthLease related to it,
+        it can only have PENDING or EXPIRED statuses.
+
+        With a lease, it can have any of the other statuses.
+        """
+        allowed_statuses_without_lease = [
+            ApplicationStatus.PENDING,
+            ApplicationStatus.EXPIRED,
+        ]
+
+        if (
+            not hasattr(self, "lease")
+            and self.status not in allowed_statuses_without_lease
+        ):
+            raise ValidationError(
+                _(
+                    "BerthApplication with no lease can only be "
+                    f"{', '.join([status.name for status in allowed_statuses_without_lease])}; "
+                    f"has {self.status.name}"
+                )
+            )
+
+    def save(self, *args, **kwargs):
+        # Ensure clean is always ran
+        self.clean()
```

Comment:
you should call `full_clean`. Django explains [here](https://docs.djangoproject.com/en/3.0/ref/models/instances/#validating-objects) why.